### PR TITLE
Local docker-compose dev file

### PR DESCRIPTION
**Pull request by markbastian**
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/28173)

### CODE REVIEW

File: deps.edn [hunk 1]
```diff
@@ -489,6 +489,9 @@
   {:extra-deps {nrepl/nrepl {:mvn/version "0.9.0"}}
    :main-opts  ["-m" "nrepl.cmdline"]}
 
+  ;; - start a Socket REPL on port 50505 that you can connect your editor to:
+  :socket {:jvm-opts ["-Dclojure.server.repl={:address,\"0.0.0.0\",:port,50505,:accept,clojure.core.server/repl}"]}
+
```
Comment (markbastian) [inline deps.edn:494]: Just adding this so we can easily jack in.

**Comment by gusaiani** [comment]
It might be very useful to document usage of this as if you were an onboarding FE engineer, that is, with a lot of hand-holding.

**Review by metamben — APPROVED** [review]
I think this is a good start.

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+      - "13306:3306"
+    restart: always
+    profiles:
+      - mysql8data
+    networks:
+      - metabasenet
+
+  # Plain old mongo db + UI services
+  mongo:
+    image: mongo
```
Comment (metamben) [inline docker-compose-dev.yml:170]: Could we use `metabase/qa-databases:mongo-sample-...` instead?

Comment (markbastian) [reply docker-compose-dev.yml]: I changed it to use one of our mongo qa images.

**Pull request merged**